src/main/java/sda/training/rps/app/GameService.java

```diff
@@ -1,136 +1,317 @@
+    private boolean keepPlaying() {
+        System.out.print("Czy chcesz kontynuować?(y/n): ");
+        String playerChoice = scanner.scannerLine();
+        switch (playerChoice.toLowerCase()) {
+            case "n":
+            case "no":
+            case "nie":
+                return false;
+            default:
+                return true;
+        }
+    }
+
+    private boolean isContinuable(boolean continueGame) {
+        return game.getPlayerScore() < game.getWinStagesNo() &&
+                game.getComputerScore() < game.getWinStagesNo() &&
                 continueGame;
     }
 
     private void addPoints(Result result) {
+        if (addPointsConditions(result)) {
+            game.setPlayerScore(game.getPlayerScore() + result.getPlayerPoint());
+            game.setComputerScore(game.getComputerScore() + result.getComputerPoint());
+        }
     }
 
-    @Override
+    private boolean addPointsConditions(Result result) {
+        return result != null && game.getPlayerScore() != null && game.getComputerScore() != null;
+    }
+
+    private boolean isFinished() {
+        return game.getResult() != null;
+    }
+
+
     public void loadOldGame(Player player) {
         showCustomeMessage();
-        List<Game> games = loadListOfOldGames(player);
-        if (games != null) {
-            games.forEach(System.out::println);
+        List<Game> games = gameDao.findAllNotCompleteOfPlayer(player);
+
+        if (!games.isEmpty()) {
+            games.forEach(this::loadScores);
+            games.forEach(oldGame ->
+                    System.out.println((games.indexOf(oldGame) + 1)
+                            + ".\t"
+                            + printGame(oldGame))
+            );
         } else {
-            System.out.println();
+            System.out.println("Brak niedokończonych gier");
             return;
         }
+
         int choice = getPlayerChoice(games.size());
-        game = games.get(choice);
-        if (game.getPlayerScore() == 0 &&
-                game.getComputerScore() == 0) {
-            loadScores();
+        if (choice == -1) {
+            return;
         }
+        game = games.get(choice);
+
+        System.out.println("Wybrano grę:");
+        System.out.println(printGame(game));
+
         loopStages();
     }
 
-    private void showCustomeMessage() {
+    private String printGame(Game game) {
+        return "Ilość koniecznych wygranych: " + game.getWinStagesNo()
+                + " Punkty gracza: " + game.getPlayerScore()
+                + " Punkty komputera: " + game.getComputerScore()
+                + " Data rozpoczęćia: " + game.getStartDate();
     }
 
+    private void showCustomeMessage() {
+        System.out.println("Lista niedokończonmych gier\n" +
+                "Wprowadź numer gry którą chcesz kontynuować");
 
-    private List<Game> loadListOfOldGames(Player player) {
-        return Collections.emptyList();
     }
 
-
     private int getPlayerChoice(int size) {
```

Comment:
np ta funkcje dosyc latwo wyrzucic do innej klasy :) w koncu to player choice a nie game choice